node/src/components/consensus/tests/mock_proto.rs

```diff
@@ -184,12 +191,20 @@ where
                     .pop_front()
                     .expect("should have pending blocks when handling FinalizeBlock");
                 assert!(valid, "finalized block must be validated first");
+                let height = self.finalized_blocks.len() as u64;
+                let rewards = if timestamp >= self.start_time + self.era_length.0
+                    && height + 1 >= self.era_length.1
+                {
+                    Some(Default::default())
+                } else {
+                    None
+                };
```

Comment:
Are we going to test the test code? :grimacing:

Reply:
Not sure; the alternative to this is to somehow explicitly tell the mock protocol that a particular block is the last one in the era. But I think basing that on height and timestamps is pretty natural, and not Highway-specific.